main.c

```diff
@@ -16,194 +19,265 @@
  *
  *	- parses through initial command line
  *	- in the absence of a file to parse, open main page, which can either do one of 4 things:
- *		- load a new dictionary (will replace the one that is parsed, if
+ *		- load a new dictionary (will replace the one that is parsed)
  *		- quit
- *		- 
- *
- *	- launch either interactive or batch mode
- *	- save file
+ *		- launch either interactive or batch mode
+ *		- save file
  *
  *  Sample inputs:
  *
  *  One input
  *	> ./spellcheck: opens main page
  *
  *  Two inputs
- *	> Not possible to determine whether entered filename is for dict or editing
- *	> Print error messages and exit
-
- *  Three inputs
- *	> ./spellcheck -d [dictname.txt]: stores 
+ *	> ./spellcheck [filename.txt]
+ *
+ *	Other flags:
+ *	> -d [dictname.txt]: changes dictionary
+ *
+ *	> -i/-q : interactive, quiet mode
  *
- *	> ./spellcheck -i [filename.txt]: interactive mode
+ *  > --show-suggestions: verbose mode
  *
- *	> ./spellcheck -q [filename.txt]: quiet mode
+ *	> -s [filename.txt]: file saving destination 
  *
- *	> ./spellcheck -v [filename.txt]: verbose mode
+ *	> -c : colored strings
+ *
+ *  > -v : logging
  *
  *  ...
  *  and so on, with different combinations.
  *
- *  Largest possible number of argc: 7
- *	> ./spellcheck -d my_dict.txt -q misspelled.txt -s savefilename.txt
- *
+ *  Largest possible number of argc: 9
+ *	> ./spellcheck file.txt -d my_dict.txt -q misspelled.txt -s savefilename.txt -c
  */
 
-char* modename(int mode)
-{
-	switch (mode) {
+char *modename(int *mode) {
+	switch (*mode) {
 		case QUIET_MODE:
-			return "quiet batch mode"; 
+            return "Quiet Batch Mode"; 
 		case VERBOSE_MODE:
-			return "verbose batch mode";
+            return "List Batch Mode";
 		case INTERACTIVE_MODE:
-			return "interactive mode";
+            return "Interactive Mode";
 		default:
-			break;
+            break;
 	}
-	return "quiet batch mode";
+
+	return "Interactive Mode";
 }
 
-int main(int argc, char **argv)
-{
+void change_log_level(int v) {
+    // Set log level
+    switch (v) {
+        case 0: // Only fatal
+            log_set_level(LOG_FATAL);
+            break;
+        case 1: // Warn, Error
+            log_set_level(LOG_ERROR);
+            break;
+        case 2: // Debug, Info
+            log_set_level(LOG_INFO);
+            break;
+        case 3: // Trace
+            log_set_level(LOG_TRACE);
+            break;
+        default:
+            log_set_level(LOG_FATAL);
+    }
+}
+
+int main(int argc, char *argv[]) {
+    if (argc > 9) {
+        log_error("Too many command line arguments was input.");
+        shell_usage(false);
+        log_info("Usage help page provided.");
+        exit(1);
+    }
+
 	// filenames up to 400 char
-	char* dict_name = malloc(401 * sizeof(char*));
-	char* file_name = malloc(401 * sizeof(char*));
-	char* save_file = malloc(401 * sizeof(char*));
+    char *dict = malloc(401 * sizeof(char *));
```

Comment:
UNIX has a maximum path length of 4096 characters, let's use that limit here, and `#define UNIX_MAX_PATH 4096`.

Reply:
done!